[Smarty] add a note to smarty debug feature (#368)
vanilla/docs: content/developer/smarty/index.md

### Accessing Controller Data with Smarty

You can output the content of the current page's controller's data array using Smarty by using `{$ArrayKey.NestedArrayKey}` syntax. For instance, `{$BodyClass}` in Vanilla's master view outputs the css class names for the current page. `{$User.Name}` would render the session's username. To see what data you have available to you on any given page, you can insert a `{debug}` tag into your Smarty template and a popup listing the data in the data array will appear.

> (Be very careful not to leave this feature on, as it discloses personal information like email and hashed passwords).

Reviewer: ```suggestion

**Be very careful not to leave this feature on, as it can disclose personal information like email addresses.**
```
